File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var minLength = function (s) {
    let stack = []
    for (let i in s) {
        if (s[i] == "B" && stack[stack.length - 1] == "A") {
            stack.pop()
            i++
            // break
        }
        else if (s[i] == "D" && stack[stack.length - 1] == "C") {
            stack.pop()
            i++
        }
        else {
            stack.push(s[i])
        }
    }
    return stack.length
};

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        if len(s) == 1:
            return 1
        for i in range(len(s)):
            if s[i] == "B" and stack and stack[-1] == "A":
                stack.pop()
                i += 1
            elif s[i] == "D" and stack and stack[-1] == "C":
                stack.pop()
                i += 1
            else:
                stack.append(s[i])
        return len(stack)